application-xpoll-api/src/main/java/com/xwiki/xpoll/internal/DefaultXPollManager.java

```diff
@@ -195,21 +213,11 @@ private void updateWinner(XWikiContext context, XWikiDocument doc) throws XWikiE
         return currentWinners;
     }
 
-    private Map<String, Integer> getXPollResults(List<BaseObject> xpollVotes, List<String> proposals)
+    private Map<String, Integer> getXPollResults(List<BaseObject> xpollVotes, List<String> proposals, String pollType)
     {
-        Map<String, Integer> voteCount = new HashMap<>();
-        for (String proposal : proposals) {
-            voteCount.put(proposal, 0);
-        }
-        for (BaseObject xpollVote : xpollVotes) {
-            List<String> currentVotes = xpollVote.getListValue(VOTES);
-            for (String currentVote : currentVotes) {
-                if (proposals.contains(currentVote)) {
-                    int nbvotes = voteCount.get(currentVote) + 1;
-                    voteCount.put(currentVote, nbvotes);
-                }
-            }
+        if ("2".equals(pollType)) {
```

Comment:
The poll type should be a role hint (e.g. "condorcet"). You will then use the component manager to look up the ``PollResultsCalculator`` implementation with the specified hint or the default implementation.